cmd/nodecmd/create.go

```diff
@@ -0,0 +1,678 @@
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/spf13/cobra"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func newCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [subnetName]",
+		Short: "Create a new subnet configuration",
+		Long: `The subnet create command builds a new genesis file to configure your Subnet.
+By default, the command runs an interactive wizard. It walks you through
+all the steps you need to create your first Subnet.
+
+The tool supports deploying Subnet-EVM, and custom VMs. You
+can create a custom, user-generated genesis with a custom VM by providing
+the path to your genesis and VM binaries with the --genesis and --vm flags.
+
+By default, running the command with a subnetName that already exists
+causes the command to fail. If you’d like to overwrite an existing
+configuration, pass the -f flag.`,
+		SilenceUsage: true,
+		Args:         cobra.ExactArgs(1),
+		RunE:         createNode,
+	}
+
+	return cmd
+}
+
+func getNewKeyPairName(ec2Svc *ec2.EC2) (string, error) {
+	ux.Logger.PrintToUser("What do you want to name your key pair?")
+	for {
+		newKeyPairName, err := app.Prompt.CaptureString("Key Pair Name")
+		if err != nil {
+			return "", err
+		}
+		keyPairExists, err := checkKeyPairExists(ec2Svc, newKeyPairName)
+		if err != nil {
+			return "", err
+		}
+		if !keyPairExists {
+			return newKeyPairName, nil
+		}
+		ux.Logger.PrintToUser(fmt.Sprintf("Key Pair named %s already exists", newKeyPairName))
+	}
+}
+
+func removeExistingTerraformFiles() error {
+	nodeConfigFile := "node_config.tf"
+	terraformLockFile := ".terraform.lock.hcl"
+	terraformStateFile := "terraform.tfstate"
+	terraformStateBackupFile := "terraform.tfstate.backup"
+	if _, err := os.Stat(nodeConfigFile); err == nil {
+		err := os.Remove(nodeConfigFile)
+		if err != nil {
+			return err
+		}
+	}
+	if _, err := os.Stat(terraformLockFile); err == nil {
+		err := os.Remove(terraformLockFile)
+		if err != nil {
+			return err
+		}
+	}
+	if _, err := os.Stat(terraformStateFile); err == nil {
+		err := os.Remove(terraformStateFile)
+		if err != nil {
+			return err
+		}
+	}
+	if _, err := os.Stat(terraformStateBackupFile); err == nil {
+		err := os.Remove(terraformStateBackupFile)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getCertFilePath(certName string) (string, error) {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+	certFilePath := homeDir + "/.ssh/" + certName
+	return certFilePath, nil
+}
+
+func createNodeConfig(nodeID, region, ami, keyPairName, certPath, sg, eip, clusterName string) error {
+	elasticIPToUse := eip[1 : len(eip)-2]
+	nodeIDToUse := nodeID[1 : len(nodeID)-2]
+
+	nodeConfig := models.NodeConfig{
+		NodeID:        nodeIDToUse,
+		Region:        region,
+		AMI:           ami,
+		KeyPair:       keyPairName,
```

Comment:
Do KeyPair and Cert refer to the same thing? If this is the case can we unify naming?